### PR TITLE
Update cab

### PR DESCRIPTION
**Pull request by alishamohanty**
This PR solves issue #19.
The PR -> Allows a cab to be updated with new details provided(except for license plate which cannot be modified).

https://user-images.githubusercontent.com/36623306/145344420-8ead7d5f-70a1-4bbd-8bf3-e2a23ae35a8c.mp4

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -17,7 +17,7 @@
                  [org.clojure/spec.alpha "0.3.214"]
                  [migratus "1.3.5"]
                  [camel-snake-kebab "0.4.2"]
-                 [honeysql "1.0.461"]]
+                 [com.github.seancorfield/honeysql "2.1.833"]]
```
Comment (govindkrjoshi) [inline project.clj:20]: :+1:

File: src/winter_onboarding_2021/core.clj [hunk 2]
```diff
@@ -0,0 +1 @@
+(ns winter-onboarding-2021.core)
```
Comment (govindkrjoshi) [inline src/winter_onboarding_2021/core.clj:1]: Why is this necessary?

File: src/winter_onboarding_2021/fleet_management_service/README.md [hunk 3]
```diff
@@ -5,12 +5,13 @@
 1. Clone this repo `git clone https://github.com/nilenso/winter-onboarding-2021.git`
 2. `cd` into `winter-onboarding-2021`
 3. Install dependencies `lein deps`
-4. Run the migrations
+4. Create the databases: `fleet_management`, and `fleet_management_test`
```
Comment (govindkrjoshi) [inline src/winter_onboarding_2021/fleet_management_service/README.md:8]: :+1:

File: src/winter_onboarding_2021/core.clj (hunk 2 above)
Comment (alishamohanty) [reply src/winter_onboarding_2021/core.clj:1]: I was unable to start repl, because of absence of core file in fms(fleet management service)

File: src/winter_onboarding_2021/fleet_management_service/views/cab.clj [hunk 4]
```diff
@@ -71,11 +72,6 @@
          :name "name"
          :required true
          :value name)]
-       [:div {:class "mb-3"}
```
Comment (govindkrjoshi) [inline src/winter_onboarding_2021/fleet_management_service/views/cab.clj]: Instead of removing this you can disable the input.

**Review by govindkrjoshi — APPROVED** [review]

File: src/winter_onboarding_2021/fleet_management_service/handlers/cab.clj [hunk 5]
```diff
@@ -49,3 +60,20 @@
...
+          (assoc-in [:flash :data] cab)
+          (merge (response/redirect "/cabs/new")))
+      (do
+        (models/update! id validated-cab)
+        (merge update-success-flash
+               (response/redirect (format "/cabs/%s" id)))))))
+
+(defn update-cab-view [req]
+  (let [cab (models/get-by-id (get-in req [:params :id]))]
+    {:title "Update cab"
```
[the remark this answers is not in the document]
Comment (alishamohanty) [reply src/winter_onboarding_2021/fleet_management_service/handlers/cab.clj:78]: Sure, we can have. What do u thing about ```update cab - <licence-plate>``` ?

**Comment by alishamohanty** [comment]
Update: Added tests for the 'update cab view' handler. 
Made the title for update page dynamic (by adding the licence number to the title)

File: src/winter_onboarding_2021/core.clj (hunk 2 above)
Comment (govindkrjoshi) [reply src/winter_onboarding_2021/core.clj:1]: This file is not necessary. We have change the init-ns in the project clj to user.clj

Comment (alishamohanty) [reply src/winter_onboarding_2021/core.clj:1]: Ok. Shall I make the change in this PR itself?

Comment (alishamohanty) [reply src/winter_onboarding_2021/core.clj:1]: Ok, cool. Will remove it.

File: src/winter_onboarding_2021/fleet_management_service/handlers/cab.clj (hunk 5 above)
[the remark this answers is not in the document]
Comment (shhivam) [reply src/winter_onboarding_2021/fleet_management_service/handlers/cab.clj]: sure
